website/static/js/profile.js

```diff
@@ -468,7 +468,44 @@ var NameViewModel = function(urls, modes, preventUnsaved, fetchCallback) {
         return suffix;
     };
 
+    self.fullNameParsed = function() {
```

Comment:
This should be a module on its own, either as a function in osfHelpers or completely separate, and fully tested. There are too many edge cases to hope it handles them all properly.